### PR TITLE
[MIG] 9.0 - auditlog

### PR DESCRIPTION
**Pull request by sebalix**
Supersedes #454 (no response)

Migration done following the instructions [from the wiki](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0#technical-method-to-migrate-a-module-from-80-to-90-branch) + Cherry-picked the commits done by @holdenrehg in #454.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Thanks for the followup. Please squash all the "OCA Transbot" commits in one, and check Travis status.

**Comment by sebalix** [comment]
Module tested on runbot (full log, fast log, user session, HTTP requests...), works like expected.

Travis fails on the `database_cleanup` module.

**Comment by sebalix** [comment]
As soon as this module is merged I'll port it on 10.

**Comment by sebalix** [comment]
@pedrobaeza sorry I didn't see the lint errors, it's fixed now.

I also migrated the last methods to the new API, PR ready for reviews now.

File: auditlog/models/autovacuum.py
Comment (pedrobaeza) [inline auditlog/models/autovacuum.py:37]: Maybe we need to launch a real "VACUUM" here?

**Review by pedrobaeza — APPROVED** [review]
Little comment, but overall :+1:

File: auditlog/models/autovacuum.py
Comment (sebalix) [reply auditlog/models/autovacuum.py:37]: Why not, but I would prefer in a new PR as it's a new feature.
About the VACUUM, I was thinking that Odoo already performs this operation through `ir.autovacuum` https://github.com/odoo/odoo/blob/9.0/openerp/addons/base/ir/ir_autovacuum.py but it's not a real VACUUM neither.

**Review by lasley — APPROVED** [review]
Thanks @sebalix - I agree with your eval that adding the actual vacuum should be a separate PR, which would also allow for easy backports

**Comment by lasley** [comment]
Travis is 🔴  but not on your module and I think it's been resolved already. Restarting build; this can be merged if 🍏

**Comment by sebalix** [comment]
PR rebased but it's failing on `database_cleanup`.

**Comment by pedrobaeza** [comment]
This PR is supposed to fix it: https://github.com/OCA/server-tools/pull/612

**Comment by sebalix** [comment]
Rebased to include #612 , Travis is now green

**Comment by pedrobaeza** [comment]
OK, merging. Thanks for the PR.

**Pull request merged**
